rama-tls/src/boring/client/connector.rs

```diff
@@ -437,9 +438,24 @@ impl<S, K> TlsConnector<S, K> {
                 if let Some(ref proto) = application_layer_protocol {
                     tracing::trace!(%proto, "boring client (connector) has selected ALPN");
                 }
+
+                let store_server_cert_chain = connector_data
+                    .is_some_and(|data| data.connect_config_input.store_server_certificate_chain);
+
+                let server_certificate_chain = store_server_cert_chain
+                    .then(|| {
+                        stream
+                            .ssl()
+                            .peer_cert_chain()
+                            .map(|chain| chain.try_into().ok())
+                    })
+                    .flatten()
```

Comment:
you might be trying to be a bit too clever with the functional style here. E.g. using it on booleans in this way seems to force you to flatten twice. Might want to write it a bit different